.github/workflows/main.yml

```diff
@@ -0,0 +1,82 @@
+name: taisei-build-test
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.yml"
+      - "**.yaml"
+  pull_request:
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.yml"
+      - "**.yaml"
+
+jobs:
+  linux-test-build:
+    runs-on: ubuntu-20.04 # so we're not using ancient tools
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+
+    - name: Checkout Submodules
+      run: git submodule update --init --recursive
+
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    # needed because apt version is too old
+    - name: Install Build Tools (pip)
+      run: pip install meson ninja
```

Comment:
Ubuntu 20.04 should have meson 0.53.2, which should be new enough... if that doesn't work, we need to fix our build system.

Reply:
Yep, missed that. It was from before when I was testing with Ubuntu 18.04, which had an old `meson`. I bumped it up to 20.04 since that's going to become the GitHub standard in a month anyways.